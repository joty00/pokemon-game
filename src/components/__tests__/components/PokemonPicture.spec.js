import { describe, expect, test } from "vitest";
import { shallowMount } from "@vue/test-utils";
import PokemonPicture from "@/components/PokemonPicture.vue"


describe('PokemonPicture component', () => {

    test('debe de hacer match con el sanpshot ', () => {
        const wrapper = shallowMount(PokemonPicture, {
            props: {
                pokemonId: 1,
                showPokemon: false
            }
        })
        expect(wrapper.html()).toMatchSnapshot()
    })


    test('debe mostrar la imagen oculta y el pokemon 100', () => {
        const wrapper = shallowMount(PokemonPicture, {
            props: {
                pokemonId: 100,
                showPokemon: false
            }
        })

        const [img1, img2] = wrapper.findAll('img')

        expect(img1.exists()).toBeTruthy()
        expect(img2).toBe(undefined)

        expect(img1.classes('hidden-pokemon')).toBeTruthy()
    })


    test('debe mostrar la imagen de pokemon si showPokemon is true', () => {
        const wrapper = shallowMount(PokemonPicture, {
            props: {
                pokemonId: 100,
                showPokemon: true
            }
        })

        const img1 = wrapper.find('img')

        expect(img1.exists()).toBeTruthy()
        expect(img1.classes('hidden-pokemon')).toBe(false)
        expect(img1.classes('fade-in')).toBe(true)
        expect(img1.attributes('src')).toBe('https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg')
    })


})