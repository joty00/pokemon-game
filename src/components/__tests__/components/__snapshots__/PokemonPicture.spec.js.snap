// Vitest Snapshot v1

exports[`PokemonPicture component > debe de hacer match con el sanpshot  1`] = `"<div data-v-c76ad58e=\\"\\" class=\\"pokemon-container\\"><img data-v-c76ad58e=\\"\\" src=\\"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg\\" class=\\"hidden-pokemon\\" alt=\\"pokemon\\"></div>"`;
